.github/workflows/bump-deps.yaml

```diff
@@ -0,0 +1,74 @@
+# Description: Run this by visiting https://github.com/brave/brave-core/actions/workflows/bump-deps.yaml
+# and clicking the `Run workflow` button on the top-right.
+# You can also run it from cli by: `gh workflow run bump-deps.yaml -f ref=commit-hash`
+name: Bump DEPS
+on:
+  workflow_dispatch:
+    inputs:
+      dep:
+        description: "Dependency"
+        required: true
+        type: string
+        default: "vendor/web-discovery-project"
+      ref:
+        description: "New ref for dependency"
+        required: false
+        type: string
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-bump:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: "0"
+
+      - name: auto-bump
+        run: |
+          shopt -s inherit_errexit
+          set -eEo pipefail
+
+          SED=sed
+
+          # Inputs
+          DEP="${{ inputs.dep }}"
+          DEPREF="${{ inputs.ref }}"
+
+          [[ "${DEP:?}" =~ ^[a-zA-Z/-]+$ ]] || { echo Input dependency validation failed; exit 1; }
+          if [[ -z "${DEPREF}" ]]; then
+            echo "Attempting to guess ref hash"
+            DEP_REPO=$(sed -n '/deps = {/, /^}/p' DEPS | sed 's/deps = //' | python3 -c "import sys,ast; d=ast.literal_eval(sys.stdin.read()); dep=d[\""$DEP"\"]; print(dep) if isinstance(dep, str) else print(dep.get(\"url\"))" | grep -Po '(?<=github.com\/).*?(?=\.git)?(?=@)' | sed 's/\.git//' )
```

Comment:
Let's at least break lines, this is a bit too long (and getting quite complex)
```bash
DEP_REPO=$(\
    sed -n '/deps = {/, /^}/p' DEPS|\
    sed 's/deps = //'|\
    python3 -c "import sys,ast; d=ast.literal_eval(sys.stdin.read()); dep=d[\""$DEP"\"]; print(dep) if isinstance(dep, str) else print(dep.get(\"url\"))"|\
    grep -Po '(?<=github.com\/).*?(?=\.git)?(?=@)'|\
    sed 's/\.git//')
```

Reply:
Thanks! added in [ec36246](https://github.com/brave/brave-core/pull/16068/commits/ec3624681a0ea368158e5545bb238ec3ef9042c4)